### PR TITLE
KZOO-268: Allow to configure call being recorded announcement

### PR DESCRIPTION
**Pull request by harenson**
Depends on:
- https://github.com/2600hz/kazoo-crossbar/pull/350
- https://github.com/2600hz/kazoo-sounds/pull/27
- https://github.com/2600hz/kazoo-core/pull/503

This change allows to configure at account level whether or not to play a prompt (after bridge) to announce that the call is being recorded. Intended only for the cases when call_recording is being enforced at account level (implicit call_recording).

Whenever a bridge event is received by the callflow executor (cf_exe), if `{CALL_ACCOUNT}.call_recording.account.{CALL_DIRECTION}.{CALL_ORIGIN}.should_announce_when_recording=true`, the prompt announcing call is being recorded will be played.

### CODE REVIEW

File: src/cf_exe.erl [hunk 1]
```diff
@@ -962,8 +962,53 @@ handle_channel_replaced(Call, CallEvent, Notify) ->
...
     relay_message(Notify, CallEvent).
 
+-spec maybe_announce_call_is_being_recorded(kapps_call:call()) -> 'ok'.
+maybe_announce_call_is_being_recorded(Call) ->
+    maybe_announce_call_is_being_recorded(Call, kapps_call:is_recording(Call)).
+
+-spec maybe_announce_call_is_being_recorded(kapps_call:call(), boolean()) -> 'ok'.
+maybe_announce_call_is_being_recorded(Call, 'true') ->
+    {'ok', Account} = kzd_accounts:fetch(kapps_call:account_id(Call)),
+    ShouldPlayPrompt = kz_json:is_true([kapps_call:direction(Call) %% Direction: inbound | outbound.
```
Comment (jamesaimonetti) [inline src/cf_exe.erl:976]: What about `any`?

File: src/cf_exe.erl [hunk 2]
```diff
@@ -962,8 +962,53 @@ handle_channel_replaced(Call, CallEvent, Notify) ->
...
+                                      ,kzd_accounts:call_recording_account(Account)
+                                      ),
+    maybe_announce_call_is_being_recorded(Call, 'true', ShouldPlayPrompt);
+maybe_announce_call_is_being_recorded(Call, 'false') ->
+    lager:debug("call is not being recorded, not trying to play prompt"),
+    maybe_announce_call_is_being_recorded(Call, 'false', 'false').
+
+-spec maybe_announce_call_is_being_recorded(kapps_call:call(), boolean(), boolean()) -> 'ok'.
+maybe_announce_call_is_being_recorded(Call, 'true', 'true') ->
+    PromptId = <<"call-is-being-recorded">>,
```
Comment (jamesaimonetti) [inline src/cf_exe.erl]: Prompts typically have a name like `cf-call_is_being_recorded` or similar (see for instance `vm-enter_pass`); the prefix being the app or feature where the prompt is typically used.

File: src/cf_exe.erl [hunk 3]
```diff
@@ -962,8 +962,53 @@ handle_channel_replaced(Call, CallEvent, Notify) ->
...
+    lager:debug("playing ~s prompt", PromptId),
+    CallId = kapps_call:call_id_direct(Call),
```
Comment (jamesaimonetti) [inline src/cf_exe.erl:991]: Why not simply:
```erlang
maybe_announce_call_is_being_recorded(Call, 'true', 'true') ->
    NoopId = kapps_call_command:prompt(<<"cf-call_is_being_recorded">>, Call),
    lager:info("playing recording prompt, waiting for ~s", [NoopId]),
    _Wait = kapps_call_command:wait_for_noop(Call, NoopId),
    lager:info("prompt has finished playing: ~p", [_Wait]);
```
Or `b_prompt` and ignore the noop tracking?

File: src/cf_exe.erl [hunk 4]
```diff
@@ -962,8 +962,53 @@ handle_channel_replaced(Call, CallEvent, Notify) ->
...
+    Prompt = kapps_call:get_prompt(Call, PromptId),
+    Commands = [kz_json:from_list([{<<"Application-Name">>, <<"noop">>}
```
Comment (jamesaimonetti) [inline src/cf_exe.erl:993]: Not sure what the queue does here in any case; the noop gets executed, then the prompt is played...what purpose did the noop have here?

**Review by jamesaimonetti — CHANGES_REQUESTED** [review]

File: src/cf_exe.erl (hunk 1 above)
Comment (harenson) [reply src/cf_exe.erl:976]: Fixed by exporting and using `kz_endpoint:merge_call_recording/1` function.

File: src/cf_exe.erl (hunk 2 above)
Comment (harenson) [reply src/cf_exe.erl]: Renamed to `general-call_is_being_recorded`

File: src/cf_exe.erl (hunk 3 above)
Comment (harenson) [reply src/cf_exe.erl:991]: It needs to have the `Insert-At=now` header, so the prompt gets played right away.

File: src/cf_exe.erl (hunk 4 above)
Comment (harenson) [reply src/cf_exe.erl:993]: Me neither, I just copied `kapps_call_command:play/5` functionality.

**Pull request merged**